lib/galaxy/tools/__init__.py

```diff
@@ -3038,12 +3038,89 @@ def produce_outputs(self, trans, out_data, output_collections, incoming, history
+        input_length = hdca.collection.element_count
+
+        if how_split == "maximum_elements":
+            iter_list = iter(hdca.collection.elements)
+            number_of_collections_out = input_length // number
+
+            # If the input collection is not completely divisible by the integer input, need one more collection out
+            if (input_length % number != 0):
+                number_of_collections_out += 1
+
+            for i in range(0, number_of_collections_out):
+                current_collection_elements = odict()
+
+                for j, dce in enumerate(iter_list, start=1):
+                    dce_object = dce.element_object
+                    element_identifier = dce.element_identifier
+
+                    if getattr(dce_object, "history_content_type", None) == "dataset":
+                        copied_value = dce_object.copy(force_flush=False)
+                    else:
+                        copied_value = dce_object.copy()
+
+                    current_collection_elements[element_identifier] = copied_value
+                    if (j >= number):
+                        break
+
+                self._add_datasets_to_history(history, itervalues(current_collection_elements))
+                output_collections.create_collection(
+                    self.outputs["output"], "output_{}".format(i), elements=current_collection_elements
+                )
+        else:
+            small = input_length // number
+            big = small + 1
+
+            # Number of each group:
+            num_big = input_length % number  # The remainder
+            num_small = number - num_big
+
+            result = []
+
+            # Build result:
+            for i in range(0, num_big):
+                result.append(hdca.collection.elements[i * big: i * big + big])
+
+            offset = num_big * big
+            for i in range(0, num_small):
+                result.append(hdca.collection.elements[offset + i * small: offset + i * small + small])
+
+            # Results to history
+            for i, collection in enumerate(result):
+                if len(collection) == 0:
+                    break
+                current_collection_elements = odict()
+
+                for dce in collection:
+                    dce_object = dce.element_object
+                    element_identifier = dce.element_identifier
+
+                    if getattr(dce_object, "history_content_type", None) == "dataset":
+                        copied_value = dce_object.copy(force_flush=False)
+                    else:
+                        copied_value = dce_object.copy()
+                    current_collection_elements[element_identifier] = copied_value
+
+                self._add_datasets_to_history(history, itervalues(current_collection_elements))
+                output_collections.create_collection(
+                    self.outputs["output"], "output_{}".format(i), elements=current_collection_elements
```

Comment:
The problem here is that you're dynamically creating outputs (`output_0`, `output_1`) etc.
This makes it hard to predict how many outputs will be generated ahead of time, and these dynamically generated outputs won't be connectable in the workflow editor. That doesn't necessarily mean we don't want this, but I just wanted to point this out.

Reply:
Yes that is hopefully not a problem, the main goal/factor of this was to allow someone to do an initial analysis on say 200 samples and then split that collection into 4 to run a more computationally demanding analysis on 50 of them without needing to reupload the samples

Reply:
Yeah - I never intended for this to work this way. It is concerning, I'm not sure it is wrong though.